content/en/tracing/trace_collection/library_config/nodejs.md

```diff
@@ -26,87 +26,211 @@ Tracer settings can be configured with the following environment variables:
+**Example**: `mysql:my-mysql-service-name-db,pg:my-pg-service-name-db`<br>
 Provide service names for each plugin. Accepts comma separated `plugin:service-name` pairs, with or without spaces.
 
 `DD_TRACE_DISABLED_PLUGINS`
-: A comma-separated string of integration names automatically disabled when the tracer is initialized. Environment variable only, for example, `DD_TRACE_DISABLED_PLUGINS=express,dns`.
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+**Example**: `DD_TRACE_DISABLED_PLUGINS=express,dns`<br>
+A comma-separated string of integration names automatically disabled when the tracer is initialized.
 
 `DD_TRACE_LOG_LEVEL`
-: **Default**: `debug`<br>
+: **Configuration**: `logLevel`<br>
+**Default**: `debug`<br>
 A string for the minimum log level for the tracer to use when debug logging is enabled, for example, `error`, `debug`.
 
-
-For more options including the programmatic configuration API, see the [API documentation][2].
+Flush Interval
+: **Configuration**: `flushInterval`<br>
+**Default**: `2000`<br>
+Interval in milliseconds at which the tracer will submit traces to the agent.
+
+`DD_TRACE_PARTIAL_FLUSH_MIN_SPANS`
+: **Configuration**: `flushMinSpans`<br>
+**Default**: `1000`<br>
+Number of spans before partially exporting a trace. This prevents keeping all the spans in memory for very large traces.
+
+`DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP`
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+A regex to redact sensitive data from incoming requests' query string reported in the `http.url` tag (matches will be replaced with `<redacted>`). Can be an empty string to disable redaction or `.*` to redact all query string. **WARNING: this regex will execute for every incoming request on an unsafe input (url), please make sure you use a safe regex.** |
+
+`DD_TRACE_CLIENT_IP_HEADER`
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+Custom header name to source the `http.client_ip` tag from.
+
+DNS Lookup Function
+: **Configuration**: `lookup`<br>
+**Default**: `require('dns').lookup`<br>
+Custom function for DNS lookups when sending requests to the agent.
+
+`DD_TRACE_AGENT_PROTOCOL_VERSION`
+: **Configuration**: `protocolVersion`<br>
+**Default**: `0.4`<br>
+Protocol version to use for requests to the agent. The version configured must be supported by the agent version installed or all traces will be dropped.
+
+`DD_PROFILING_ENABLED`
+: **Configuration**: `profiling`<br>
+**Default**: `false`<br>
+Whether to enable profiling.
+
+`DD_TRACE_REPORT_HOSTNAME`
+: **Configuration**: `reportHostname`<br>
+**Default**: `false`<br>
+Whether to report the system's hostname for each trace. When disabled, the hostname of the agent will be used instead.
+
+Experimental Features
+: **Configuration**: `experimental`<br>
+**Default**: `{}`<br>
+Experimental features can be enabled all at once using boolean `true` or individually using key/value pairs. Please contact us to learn more about the available experimental features.
```

Comment:
I feel like a lot of users will probably just blindly enable all without talking to us. Should be document this at all?

Reply:
I suppose it's up to the team. There's no hint as to what the features are in this documentation page but users can certainly find them in the codebase.

Reply:
Well, by the sentence it sounds like they can just do `experimental: true` to turn on _everyting_. That's what worries me. I think many users will just turn everything on without even looking. 😬

Reply:
Ah, you're right, whoever wrote that text originally intended `experimental:true` to work and to set each individual flag. I agree that it would be dangerous.

I looked through the code and I couldn't see how it could work. So I ran the code and set that flag to `true` and none of the experiments were enabled. I'll update this to remove the text about enabling all experiments.